lib/utils/compose.coffee

```diff
@@ -349,13 +349,42 @@ tagServiceImages = (docker, images, serviceImages) ->
 			logs: d.logs
 			props: d.props
 
-authorizePush = (tokenAuthEndpoint, registry, images) ->
+
+getPreviousRepos = (logger, appID) ->
+	sdk = require('balena-sdk').fromSharedOptions()
+	DockerToolbelt = require('docker-toolbelt')
+	dockerToolbelt = new DockerToolbelt()
+	sdk.pine.get(
```

Comment:
This request pulls all releases on the app which can be fairly expensive — it could be hundreds of rows. Might be better to sort by id descending and limit to the latest 10 or so results. The Pine options you’re looking for to do this are iirc “$order” and “$top”.

Reply:
You also don’t need any fields on the release so you could also limit what is fetched with “$select: [ “id” ]”

Reply:
Nice, thanks!